cmake/ProjectUVWASI.cmake

```diff
@@ -0,0 +1,53 @@
+if(ProjectUVWASIIncluded)
+    return()
+endif()
+set(ProjectUVWASIIncluded TRUE)
+
+include(ExternalProject)
+
+set(prefix ${CMAKE_BINARY_DIR}/_deps)
+set(source_dir ${prefix}/src/uvwasi)
+set(binary_dir ${prefix}/src/uvwasi-build)
+set(include_dir ${source_dir}/include)
+set(uvwasi_library ${binary_dir}/${CMAKE_STATIC_LIBRARY_PREFIX}uvwasi_a${CMAKE_STATIC_LIBRARY_SUFFIX})
+set(uv_library ${binary_dir}/_deps/libuv-build/${CMAKE_STATIC_LIBRARY_PREFIX}uv_a${CMAKE_STATIC_LIBRARY_SUFFIX})
```

Comment:
How is this library built?

Reply:
You added this line :wink:

It is needed because uvwasi builds libuv but doesn't link it into the library (could upstream it). We don't want to build another libuv ourselves due to potential version incompatibility.

Reply:
Both are static libraries, so you cannot link uv.a to uvwasi.a.

Answering my question: uvwasi downloads source code of uv using git and git tag, then builds it as a part of uvwasi CMake project.

Reply:
> Both are static libraries, so you cannot link uv.a to uvwasi.a.

If you only consider "linking" as creating a shared library, yes. However `.a` is just an archive so one can certainly "link" aka merge two archives.